servant-server/src/Servant/Server/Experimental/Auth.hs

```diff
@@ -31,13 +32,21 @@ import           Servant.Server.Internal.RoutingApplication (addAuthCheck,
                                                              withRequest)
 import           Servant.Server.Internal.ServantErr         (Handler)
 
+import           GHC.TypeLits
+
 -- * General Auth
 
 -- | Specify the type of data returned after we've authenticated a request.
 -- quite often this is some `User` datatype.
 --
 -- NOTE: THIS API IS EXPERIMENTAL AND SUBJECT TO CHANGE
-type family AuthServerData a :: *
+-- type family AuthServerData a :: *
+class HasAuthServerData a where
+
+type AuthServerData a :: *
+
+instance {-# OVERLAPPING #-} HasAuthServerData a where
+   type AuthServerData a = TypeError ('Text "aha this should never occur")
```

Comment:
Presumably CPP for GHC < 8, and a type error message? Could this be tested with doctest?